stages/org.osbuild.grub2

```diff
@@ -120,33 +129,59 @@ def main(tree, options):
     if isinstance(legacy, bool) and legacy:
         legacy = "i386-pc"
 
+    # Check if hybrid boot support is requested, i.e. the resulting image
+    # should support booting via legacy and also UEFI. In that case the
+    # canonical grub.cfg and the grubenv will be in /boot/grub2. The ESP
+    # will only contain a small config file redirecting to the one in
+    # /boot/grub2 and will not have a grubenv itself.
+    hybrid = uefi and legacy
```

Comment:
Just a thought: could we just set this to true? So even if no legacy is supported, do the redirect into /boot/grub2? And then eventually do the same in the package perhaps? Just for the sake of having fewer combinations?

Reply:
Yeah totally. And I think @martinezjavier would agree too. There are two reasons I didn't do it yet: 1) I thought the goal was for now to be close to the original images and 2) there is some weird small extra delay (similar to the one mentioned in #198) compared to just uefi boot, that I want to investigate.

Reply:
Ah, sounds good. Let's revisit this later (also after checking with Javier). I would like us to not modify things compared to the rpms more than necessary, but I think we are not under any obligation to be particularly close to the original images on stuff like this.

Reply:
> Ah, sounds good. Let's revisit this later (also after checking with Javier). I would like us to not modify things compared to the rpms more than necessary, but I think we are not under any obligation to be particularly close to the original images on stuff like this.

I very much would like to have the hybrid boot support in both the Fedora GRUB package and Anaconda. in fact this all started from one of the conversations we had with @gicmo when talking about booting in x86.

One problem is that according to @jhlavac there are some older BIOS that only are able to detect a MBR and not a GPT. He should be able to provide more info about this.

But that doesn't mean that we couldn't implement some of the improvements in this PR, for example using the GRUB config that's in the boot directory for EFI and only having a redirect config in the ESP is something that could be done as a first step even without the hybrid boot support.